### PR TITLE
Bump golangci-lint to v1.20.0

### PR DESCRIPTION
**Pull request by corneliusweig**
This version of golangci-lint comes with significant memory improvements, making some
adjustments for small memory footprint in the CI obsolete.

See https://github.com/golangci/golangci-lint/issues/337

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -51,13 +51,13 @@ MINIKUBE_RELEASES_URL=https://github.com/kubernetes/minikube/releases/download
 
 KERNEL_VERSION ?= 4.15
 # latest from https://github.com/golangci/golangci-lint/releases
-GOLINT_VERSION ?= v1.18.0
+GOLINT_VERSION ?= v1.20.0
 # Limit number of default jobs, to avoid the CI builds running out of memory
 GOLINT_JOBS ?= 4
 # see https://github.com/golangci/golangci-lint#memory-usage-of-golangci-lint
-GOLINT_GOGC ?= 8
+GOLINT_GOGC ?= 100
```
Comment (medyagh) [inline Makefile]: what is the reason for increasing 8 to 100 ?

**Comment by medyagh** [comment]
Thanks for making this update ! I see the lint already found some new lint issues that was ignored by the previous version

File: Makefile (hunk 1 above)
Comment (afbjorklund) [reply Makefile]: GOGC=100 is the default, so it is basically removing the increased garbage collection

**Comment by afbjorklund** [comment]
When I run it locally (with 4 cpus/jobs), this lowers the memory usage from 4G to 1G:

```console
$ \time make lint GOLINT_VERSION=v1.18.0
./out/linters/golangci-lint-v1.18.0 run --deadline 4m --build-tags "integration container_image_ostree_stub containers_image_openpgp" --enable goimports,gocritic,golint,gocyclo,interfacer,misspell,nakedret,stylecheck,unconvert,unparam --exclude 'variable on range scope.*in function literal|ifElseChain' ./...
62.86user 3.91system 0:20.67elapsed 322%CPU (0avgtext+0avgdata 4165192maxresident)k
582344inputs+640outputs (665major+1436497minor)pagefaults 0swaps
```
```console
$ \time make lint GOLINT_VERSION=v1.20.0
./out/linters/golangci-lint-v1.20.0 run --deadline 4m --build-tags "integration container_image_ostree_stub containers_image_openpgp" --enable goimports,gocritic,golint,gocyclo,interfacer,misspell,nakedret,stylecheck,unconvert,unparam --exclude 'variable on range scope.*in function literal|ifElseChain' ./...
Flag --deadline has been deprecated, flag will be removed soon, please, use .golangci.yml config
cmd/minikube/cmd/start.go:129:2: var `enableUpdateNotification` is unused (unused)
	enableUpdateNotification = true
	^
pkg/minikube/extract/extract.go:397:47: `checkBinaryExpression` - `e` is unused (unparam)
func checkBinaryExpression(b *ast.BinaryExpr, e *state) string {
                                              ^
pkg/minikube/tunnel/tunnel_manager.go:125:43: (*Manager).cleanup - result 0 (*k8s.io/minikube/pkg/minikube/tunnel.Status) is never used (unparam)
func (mgr *Manager) cleanup(t controller) *Status {
                                          ^
Makefile:330: receptet för målet ”lint” misslyckades
make: *** [lint] Fel 1
Command exited with non-zero status 2
185.80user 7.64system 0:54.21elapsed 356%CPU (0avgtext+0avgdata 1504696maxresident)k
829872inputs+423928outputs (925major+2657303minor)pagefaults 0swaps
```

If that is not enough for the servers, you could try something like GOGC=20 ?
(I don't know how much RAM that Travis allocates to each VM that it spins up)

https://github.com/golangci/golangci-lint#memory-usage-of-golangci-lint

I think it still takes **way** to much time to run locally (1-3 *minutes*?), though.
Similar to how I usually avoid `make test` in favor of `go test`, locally....

**Comment by afbjorklund** [comment]
Apparently it now allocates 7.5G of RAM and 2 cores:
https://docs.travis-ci.com/user/reference/overview/

And took 3 minutes (188 seconds) to run, # [8899.2](https://travis-ci.org/kubernetes/minikube/builds/59535136)
Currently takes 5 minutes to lint, and 7 minutes to test...

**Comment by tstromberg** [comment]
@corneliusweig - Thank you! Do you mind fixing the lint issues so that we don't end up with persistently failing tests?

```
cmd/minikube/cmd/start.go:129:2: var `enableUpdateNotification` is unused (unused)
	enableUpdateNotification = true
	^
pkg/minikube/extract/extract.go:397:47: `checkBinaryExpression` - `e` is unused (unparam)
func checkBinaryExpression(b *ast.BinaryExpr, e *state) string {
                                              ^
pkg/minikube/tunnel/tunnel_manager.go:125:43: (*Manager).cleanup - result 0 (*k8s.io/minikube/pkg/minikube/tunnel.Status) is never used (unparam)
func (mgr *Manager) cleanup(t controller) *Status {
                                          ^
```

**Comment by corneliusweig** [comment]
@tstromberg I fixed the new issues found by the linter. I'm pretty blown away that `unparam` recognizes  parameters as unused when only used in a recursive call to self. Wow!

@afbjorklund Thanks for doing such thorough memory usage checking. As far as I understand, there's no action needed, right?

File: Makefile (hunk 1 above)
Comment (corneliusweig) [reply Makefile]: Yes, I restored the default, but I left the link to the docs in case that setting needs to be adjusted in the future.

**Pull request merged**